client/src/components/podcasts/PodcastTile.js

```diff
@@ -1,20 +1,167 @@
-import React from "react"
+import React, { useState, useEffect } from "react"
 import { Link } from "react-router-dom"
 
 const PodcastTile = ({ podcast, user }) => {
-  let editDeleteButtons = ''
+  const [userVotes, setUserVote] = useState({
+    hasUpVoted: false,
+    hasDownVoted: false
+  })
+  const [totalVotes, setTotalVotes] = useState(0)
+
+  let upVoteButtonClass = 'button'
+  if (userVotes.hasUpVoted) {
+    upVoteButtonClass = 'success button'
+  }
+  let downVoteButtonClass = 'button'
+  if (userVotes.hasDownVoted) {
+    downVoteButtonClass = 'alert button'
+  }
+
+  const fetchVotes = async () => {
+    try {
+      const response = await fetch(`/api/v1/podcasts/${podcast.id}/votes`)
+      if (!response.ok) {
+        throw new Error(`${response.status} (${response.statusText})`)
+      }
+      const body = await response.json()
+      const votes = body.votes
+      setTotalVotes(votes.totalVotes.value)
+      if (votes.userVotes) {
+        if (votes.userVotes.value === 1) {
+          setUserVote({ hasUpVoted: true, hasDownVoted: false })
+        } else {
+          setUserVote({ hasUpVoted: false, hasDownVoted: true })
+        }
+      } else {
+        setUserVote({ hasUpVoted: false, hasDownVoted: false })
+      }
+    } catch (error) {
+      console.error(error.message)
+    }
+  }
+
+  const newVote = async (votePayload) => {
+    try {
+      const response = await fetch(`/api/v1/podcasts/${podcast.id}/votes`, {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify(votePayload)
+      })
+      if (!response.ok) {
+        throw new Error(`${response.status} (${response.statusText})`)
+      }
+      const body = await response.json()
+      const votes = body.votes
+      const voteType = votes.userVotes.value === 1 ? 'up' : 'down'
+      if (voteType === 'up') {
+        setUserVote({ hasUpVoted: true, hasDownVoted: false })
+      } else {
+        setUserVote({ hasUpVoted: false, hasDownVoted: true })
+      }
+      setTotalVotes(votes.totalVotes.value)
+    } catch (error) {
+      console.error(error.message)
+    }
+  }
```

Comment:
this is great, though it would have been allowed to combine newVote and editVote through one endpoint